cli/configdir.go

```diff
@@ -0,0 +1,136 @@
+// Copyright © 2021 Optable Technologies Inc. All rights reserved.
+// See LICENSE for details.
+package cli
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type ConfigLoader interface {
+	Unmarshal([]byte) (interface{}, error)
```

Comment:
It would be preferable if the `interface{}` was a receiving parameter instead of a returned type. This should allow (at the cost of passing the pointer to `Load|Get` methods) the usage of static typed structs instead of just dynamic `map[string]interface{}`.

Reply:
Yeah I hesitated with this approach, good point, it's also exactly compatible with the unmarshaler api.

Reply:
Fixed